public/app/features/alerting/unified/components/receivers/useAlertmanagerConfigHealth.ts

```diff
@@ -0,0 +1,38 @@
+import { AlertmanagerConfig, Route } from '../../../../../plugins/datasource/alertmanager/types';
+
+export type ContactPointConfigHealth = 'OK' | 'Unused';
+
+export interface AlertmanagerConfigHealth {
+  contactPoints: Record<string, ContactPointConfigHealth>;
+}
+
+export function useAlertmanagerConfigHealth(config: AlertmanagerConfig): AlertmanagerConfigHealth {
+  if (!config.receivers) {
+    return { contactPoints: {} };
+  }
+
+  if (!config.route) {
+    return { contactPoints: Object.fromEntries(config.receivers.map((r) => [r.name, 'Unused'])) };
+  }
+
+  const definedContactPointNames = config.receivers?.map((receiver) => receiver.name) ?? [];
+  const usedContactPoints = getUsedContactPoints(config.route);
+
+  const contactPointsHealth: AlertmanagerConfigHealth['contactPoints'] = {};
+  const configHealth: AlertmanagerConfigHealth = { contactPoints: contactPointsHealth };
+
+  definedContactPointNames.forEach((contactPointName) => {
+    contactPointsHealth[contactPointName] = usedContactPoints.includes(contactPointName) ? 'OK' : 'Unused';
```

Comment:
Can we return the number of notification policies instead of `OK`? We can map that however we want in the UI but this gives us a bit more info to work with in future designs where we might want to show how many policies rely on a particular contact point.